litho-it/src/test/java/com/facebook/litho/widget/DefaultScrollStateDetectorTest.java

```diff
@@ -0,0 +1,326 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.facebook.litho.widget;
+
+import static org.assertj.core.api.Java6Assertions.assertThat;
+
+import android.view.MotionEvent;
+import android.view.View;
+import androidx.test.core.app.ApplicationProvider;
+import com.facebook.litho.testing.testrunner.LithoTestRunner;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+/** Tests for {@link DefaultScrollStateDetector} */
+@RunWith(LithoTestRunner.class)
+public class DefaultScrollStateDetectorTest {
+
+  private View hostView;
+  private DefaultScrollStateDetector detector;
+  private TestScrollStateListener scrollStateListener;
+
+  @Before
+  public void setup() {
+    hostView = new View(ApplicationProvider.getApplicationContext());
+    scrollStateListener = new TestScrollStateListener();
+    detector = new DefaultScrollStateDetector();
+    detector.setListener(scrollStateListener);
+  }
+
+  // For scroll driven by non-fling gesture, the first onScrollChange callback triggers start event
+  // and the action_up event indicates scroll's stopping.
+  @Test
+  public void testSlowScroll() {
```

Comment:
Can I ask you to change tests naming to the scheme `method/unitUnderTest_scenario/stateUnderTest_expectedBehaviour`, please?
Like here, for example: https://github.com/facebook/litho/blob/master/litho-it/src/test/java/com/facebook/litho/LithoViewTest.java#L271-L339.

We are trying to adhere to this more descriptive pattern

Reply:
That's great practice, done!